apps/nextjs/src/hooks/useLessonPlanScrollManagement.ts

```diff
@@ -0,0 +1,155 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+
+import { LessonPlanKeys } from "@oakai/aila/src/protocol/schema";
+import { aiLogger } from "@oakai/logger";
+
+const log = aiLogger("lessons");
+
+export const useLessonPlanScrollManagement = (
+  streamingSection: LessonPlanKeys | undefined,
+  streamingSectionCompleted: LessonPlanKeys | undefined,
+) => {
+  const [recentlyScrolledSection, setRecentlyScrolledSection] = useState<
+    LessonPlanKeys | undefined
+  >(undefined);
+
+  const [userHasCancelledAutoScroll, setUserHasCancelledAutoScroll] =
+    useState(false);
+
+  const sectionRefs = useRef<
+    Record<LessonPlanKeys, React.MutableRefObject<HTMLDivElement | null>>
+  >({
+    title: { current: null },
```

Comment:
Not so happy about having these hard-coded like this, but the lesson structure seems stable, so I didn't try to make this more complex than it needed to be